Add more properties to EventSource, implement for SimTelEventSource, fixes #1286

MaxNoe opened the pull request:
For all EventSources:
* obs_id
* is_simulation
* data_levels

For SimTelEventSource:
* mc_header

watsonjj commented on ctapipe/io/simteleventsource.py:
```diff
@@ -205,6 +206,22 @@ def __exit__(self, exc_type, exc_val, exc_tb):
+
+    @property
+    def datalevels(self):
+        return ('R0', 'R1', 'DL0')
```
Instead of generic strings, would it be worth to have these as fixed enums?

watsonjj replied:
Then a developer has a reference location to specify the correct levels instead of guessing, and adds a place for documenting the levels in the context of ctapipe

MaxNoe replied:
Good Idea

MaxNoe commented:
@watsonjj I added the enum and removed two stump sources that were used nowhere (the `FitsSource` even had

```
def __iter__(self):
    # return containers
    pass
```

kosack commented on ctapipe/io/datalevels.py:
```diff
@@ -0,0 +1,13 @@
+    R2 = auto()
+    DL0 = auto()
+    DL1a = auto()
+    DL1b = auto()
```
we don't really have DL1a/b anymore. I'd just drop them for now (we realized it really depends on the use case, like training, etc, which parameters or images are filled in).  Just call it all DL1 for now until that is settled.

MaxNoe replied:
But for an event source it is a very big difference if it provides images or parameters. That was the idea behind DLa / DLb here.

Should I call it `DL1_images` ?

kosack commented on ctapipe/io/eventsource.py:
```diff
@@ -200,6 +196,40 @@ def subarray(self):
+    @property
+    @abstractmethod
+    def obs_id(self):
+        """
```
Eventually this should be a full ObservationConfiguration but since that is not well defined yet, it's ok.   That way, you get the InstrumentConfiguration (SubarrayDescription) and the ObservationConfiguration as "metadata" attached to the source.   The ObservationConfiguration would contain other parameters describing the observation mode, etc (like the mcheader for sims, which is.a sort of SimulatedObservationConfiguration), though of course they are all  keyed by the obs_id, so this is fine. 

In fact, we could make that generic: e.g. define that all EventSources provide an ObservationConfiguration (ObsConfigContainer?), and we just make MCHeader a sub-class of this, adding parameters specific to MCs.

kosack replied:
Note that you already get the obs_id inside the MCHeader (for simulations)

MaxNoe replied:
Yes, this is where I get it in the SimTelEventSource

MaxNoe commented:
@kosack @watsonjj , sorry the merge accidentally removed a needed import, tests should be passing now
